rest/rest-resources/src/main/java/org/apache/brooklyn/rest/filter/CorsImplSupplierFilter.java

```diff
@@ -0,0 +1,99 @@
+ * The right fix is to change your calling structure; architecturally, the browser shouldn't be calling the Apache Brooklyn APIs directly.
+ * A web app should be interacting solely with the Apache Brooklyn server.
+ * If there is a need to get information from Apache Brooklyn APIs then,
+ * it could either simply proxy the request or could do the request on the client's behalf and potentially further processing on the results before finally getting back to the client.
+ *
+ * If brooklyn.experimental.feature.corsCxfFeature.allowedOrigins is not supplied then allowedOrigins will be on all domains.
+ *
+ * Currently there is no fine per API request control it is rather applied to the entire server.
+ * Even if you have per API request control and apply CORS to groups of pages/resources,
+ * then you have to think about how to configure the values that get added to the CORS header,
+ * as you really don't want to use a "*" wildcard.
+ * Also you have to think about what the user interface is to capture this config, and maybe issues around how to persist it, upgrade etc.
+ *
+ * It is best when web app communicates just with its own server, not with multiple servers.
+ * It's the Apache Brooklyn server that should be the single point of contact to moderate and control access to the information from the AMP API, which should never be independently exposed to a web UI.
+ * This sort of architecture can give you additional headaches behind proxies, with firewalls, etc.
+ * For another thing, CORS can be used and can be secure enough up to a point if implemented right (and that's not a trivial 'if'),
+ * but it is still an additional attack vector that can be exploited by mitm attacks etc.
+ * In short, the proposed architecture and use of CORS is more complex, less secure,
+ * and more difficult to manage than the alternative of web client ---> Apache Brooklyn Server ----> fan out to backend servers + Apache Brooklyn API + etc.
+ *
+ * Notes by Geoff Macartney.
+ */
+@Provider
+public class CorsImplSupplierFilter extends CrossOriginResourceSharingFilter {
+    public static final ConfigKey<List<String>> ALLOWED_ORIGINS = ConfigKeys.newConfigKey(new TypeToken<List<String>>() {}, BrooklynFeatureEnablement.FEATURE_CORS_CXF_PROPERTY + ".allowedOrigins");
+    private static final Logger LOGGER = LoggerFactory.getLogger(CorsImplSupplierFilter.class);
+
+    private static final boolean brooklynFeatureEnabled = BrooklynFeatureEnablement.isEnabled(BrooklynFeatureEnablement.FEATURE_CORS_CXF_PROPERTY);
+    static {
+        if (brooklynFeatureEnabled) {
+            LOGGER.warn("CORS brooklyn feature enabled.");
+        }
+    }
+
+    public CorsImplSupplierFilter(@Nullable ManagementContext managementContext) {
+        setFindResourceMethod(false);
+        if (managementContext != null) {
+            setAllowOrigins(JavaGroovyEquivalents.<List<String>>elvis(managementContext.getConfig().getConfig(ALLOWED_ORIGINS), Collections.<String>emptyList()));
```

Comment:
It is possible for `ALLOWED_ORIGINS` to be null or empty, if so then I find elvis use full to setting it to an empty list.
`Collections.emptyList()` is used in `CrossOriginResourceSharingFilter` default constructor so I use the same class which is also immutable.